Marlin/Configuration_adv.h

```diff
@@ -3236,6 +3236,8 @@
 
 //#define REPETIER_GCODE_M360     // Add commands originally from Repetier FW
 
+//#define SAVE_CONFIG_INSIDE_FIRMWARE // This will save all user configuration options and enable M360 to dump it
```

Comment:
Here comes the controversy...

I think there is little value to this if it is not enabled by default.
The entire point is to make setting retrievable for users who do not track their changes in git, or back them up in any other way. Ideally vendors would leave it enabled as well.

If we decide to include this, I think it should be opt-out rather than opt-in. Perhaps something could be done to default-disable it on boards with very small flash memories.

We might be able to gauge the impact of this by enabling it on every example in the Config repo and testing how many now run out of space.